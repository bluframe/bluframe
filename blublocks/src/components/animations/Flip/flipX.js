// @flow

import { type KeyFrames, keyframes } from "styled-components"

const flipInX: KeyFrames = keyframes`
0% {
  -webkit-transform: perspective(400px) rotateX(90deg);
  transform: perspective(400px) rotateX(90deg);
  opacity: 0;
}

40% {
  -webkit-transform: perspective(400px) rotateX(-10deg);
  transform: perspective(400px) rotateX(-10deg);
}

70% {
  -webkit-transform: perspective(400px) rotateX(10deg);
  transform: perspective(400px) rotateX(10deg);
}

100% {
  -webkit-transform: perspective(400px) rotateX(0deg);
  transform: perspective(400px) rotateX(0deg);
  opacity: 1;
}
`

export default flipInX
