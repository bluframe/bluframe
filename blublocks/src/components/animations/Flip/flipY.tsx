import { keyframes } from "styled-components"

const flipInY = keyframes`
0% {
    -webkit-transform: perspective(400px) rotateX(90deg);
    -ms-transform: perspective(400px) rotateX(90deg);
    transform: perspective(400px) rotateX(90deg);
    opacity: 0;
  }

  40% {
    -webkit-transform: perspective(400px) rotateX(-10deg);
    -ms-transform: perspective(400px) rotateX(-10deg);
    transform: perspective(400px) rotateX(-10deg);
  }

  70% {
    -webkit-transform: perspective(400px) rotateX(10deg);
    -ms-transform: perspective(400px) rotateX(10deg);
    transform: perspective(400px) rotateX(10deg);
  }

  100% {
    -webkit-transform: perspective(400px) rotateX(0deg);
    -ms-transform: perspective(400px) rotateX(0deg);
    transform: perspective(400px) rotateX(0deg);
    opacity: 1;
  }
`

export default flipInY
