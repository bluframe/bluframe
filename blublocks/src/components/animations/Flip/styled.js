// @flow
/* eslint-disable no-ternary */

import styled, { type StyledComponent } from "styled-components"
import type { Theme } from "@bluframe/blublocks"
import flipX from "components/animations/Flip/flipX"
import flipY from "components/animations/Flip/flipY"

export const Wrapper: StyledComponent<
  {| +flip: "X" | "Y", +isAnimated: boolean |},
  Theme,
  HTMLDivElement
> = styled.div`
  animation: ${({ flip }) => (flip === "X" ? flipX : flipY)};
  animation-duration: 1s;
  animation-fill-mode: both;
  animation-play-state: ${({ isAnimated }) =>
    isAnimated ? "running" : "paused"};
`
