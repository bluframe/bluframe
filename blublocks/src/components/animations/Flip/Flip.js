// @flow

import { type ComponentProps } from "."
import React from "react"
import { Wrapper } from "./styled"

const Flip = ({ children, flip, isAnimated }: ComponentProps): React$Node => (
  <Wrapper $flip={flip} $isAnimated={isAnimated}>
    {children}
  </Wrapper>
)

export default Flip
