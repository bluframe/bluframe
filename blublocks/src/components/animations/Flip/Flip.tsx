import { type ComponentProps } from "."
import React from "react"
import { Wrapper } from "./styled"

const Flip = ({ children, flip, isAnimated }: ComponentProps) => (
  <Wrapper $flip={flip} $isAnimated={isAnimated}>
    {children}
  </Wrapper>
)

export default Flip
