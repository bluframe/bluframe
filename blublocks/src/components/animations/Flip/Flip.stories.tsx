import Flip, { type Props } from "components/animations/Flip"
import React from "react"

const stories = {
  component: Flip,
  title: "Animations/Flip"
}

const Template = (args: Props) => <Flip {...args}>{"Hello!"}</Flip>

export const Default: any = Template.bind({})
Default.args = {
  flip: "X",
  isAnimated: false
}

export const Animated: any = Template.bind({})
Animated.args = {
  flip: "X",
  isAnimated: true
}

export default stories
