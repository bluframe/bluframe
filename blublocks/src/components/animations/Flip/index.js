// @flow

import Flip from "./Flip"

export type Props = {|
  +children: React$Node,
  +flip: "X" | "Y",
  +isAnimated: boolean
|}

export type ComponentProps = {|
  +children: React$Node,
  +flip: "X" | "Y",
  +isAnimated: boolean
|}

export default Flip
