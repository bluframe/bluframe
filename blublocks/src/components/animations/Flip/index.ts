import Flip from "./Flip"
import React from "react"

export interface Props {
  children: React.ReactNode
  flip: "X" | "Y"
  isAnimated: boolean
}

export interface ComponentProps extends Props {}

export default Flip
