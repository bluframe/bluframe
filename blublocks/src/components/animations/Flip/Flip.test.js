// @flow

import Flip from "./Flip"
import React from "react"
import { render } from "tests"

jest.mock(".", () => ({}))

describe("Flip", () => {
  it("renders", () => {
    const props = {
      flip: "X",
      isAnimated: false
    }

    const { container } = render(
      <Flip {...props}>
        <div />
      </Flip>
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
