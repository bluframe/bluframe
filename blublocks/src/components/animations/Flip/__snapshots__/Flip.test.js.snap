// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flip renders 1`] = `
<div
  class="styled__Wrapper-sc-1yhc6yo-0 fLfPzz"
>
  <div />
</div>
`;
